dnf5-plugins/automatic_plugin/automatic.cpp

```diff
@@ -0,0 +1,468 @@
+        return false;
+    }
+
+    // Create a socket
+    int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    if (sockfd < 0) {
+        freeaddrinfo(server_info);
+        return false;
+    }
+
+    // Attempt to connect to the server
+    bool retval = true;
+    struct timeval timeout;
+    timeout.tv_sec = 1;
+    timeout.tv_usec = 0;
+    setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO, &timeout, sizeof(timeout));
+    if (connect(sockfd, server_info->ai_addr, server_info->ai_addrlen) < 0) {
+        retval = false;
+    }
+
+    close(sockfd);
+    freeaddrinfo(server_info);
+    return retval;
+}
+
+}  // namespace
+
+
+namespace dnf5 {
+
+void AutomaticCommand::wait_for_network() {
+    auto timeout = config_automatic.config_commands.network_online_timeout.get_value();
+    if (timeout <= 0) {
+        return;
+    }
+
+    auto & context = get_context();
+    auto & base = context.base;
+    auto & logger = *base.get_logger();
+    logger.debug("Waiting for internet connection...");
+
+    const auto time_0 = std::chrono::system_clock::now();
+    const auto time_end = time_0 + std::chrono::seconds(timeout);
+
+    // collect repository addresses to check network availability on
+    libdnf5::repo::RepoQuery repos(base);
+    repos.filter_enabled(true);
+    std::vector<std::string> urls{};
+    for (const auto & repo : repos) {
+        std::string proxy{};
+        try {
+            proxy = repo->get_config().get_proxy_option().get_value();
+            if (!proxy.empty()) {
+                urls.emplace_back(proxy);
+                continue;
+            }
+        } catch (const libdnf5::OptionValueNotSetError & e) {
+        }
+        try {
+            auto mirrorlist = repo->get_config().get_mirrorlist_option().get_value();
+            if (!mirrorlist.empty()) {
+                urls.emplace_back(std::move(mirrorlist));
+            }
+        } catch (const libdnf5::OptionValueNotSetError & e) {
+        }
+        try {
+            auto metalink = repo->get_config().get_metalink_option().get_value();
+            if (!metalink.empty()) {
+                urls.emplace_back(std::move(metalink));
+            }
+        } catch (const libdnf5::OptionValueNotSetError & e) {
+        }
+        try {
+            for (auto u : repo->get_config().get_baseurl_option().get_value()) {
+                if (!u.empty()) {
+                    urls.emplace_back(std::move(u));
+                }
+            }
+        } catch (const libdnf5::OptionValueNotSetError & e) {
+        }
+    }
+
+    // parse url to [(host, service(port number or scheme))] using libcurl
+    std::vector<std::tuple<std::string, std::string>> addresses;
+    CURLUcode rc;
+    CURLU * url = curl_url();
+    unsigned int set_flags = CURLU_NON_SUPPORT_SCHEME;
+    unsigned int get_flags = 0;
+    for (auto & u : urls) {
+        rc = curl_url_set(url, CURLUPART_URL, u.c_str(), set_flags);
+        if (!rc) {
+            std::string host{};
```

Comment:
Cosmetic - I think that `std::string host{};` could be replaced by `std::string host;`

Reply:
Done.